tcms/issuetracker/azure_boards.py

```diff
@@ -0,0 +1,142 @@
+from tcms.core.templatetags.extra_filters import markdown2html
+from tcms.issuetracker.base import IssueTrackerType, IntegrationThread
+
+
+class AzureAPI():
+    """
+        Azure Boards API interaction class.
+
+    """
+    def __init__(self, base_url=None, password=None):
+        self.api_version = "?api-version=6.0"
+        self.headers = {"Accept": 'application/json-patch+json',
+                        'Content-type': 'application/json-patch+json'}
+        self.auth = HTTPBasicAuth('apikey', password)
+        self.base_url = base_url + "/_apis/"
+
+    def get_issue(self, issue_id):
+        url = "{0}{1}{2}{3}".format(self.base_url, "wit/workitems/",
+                                    issue_id, self.api_version)
+        return self._request("GET", url, headers=self.headers, auth=self.auth)
+
+    def create_issue(self, body):
+        url = "{0}{1}{2}".format(self.base_url, "wit/workitems/$Issue", self.api_version)
+        return self._request("POST", url, headers=self.headers, auth=self.auth, json=body)
+
+    def update_issue(self, issue_id, body):
+        url = "{0}{1}{2}{3}".format(self.base_url, "wit/workitems/",
+                                    issue_id, self.api_version)
+        return self._request("PATCH", url, headers=self.headers, auth=self.auth, json=body)
+
+    def get_comments(self, issue_id):
+        headers = {"Content-type": "application/json"}
+        url = "{0}{1}{2}{3}{4}{5}".format(self.base_url, "wit/workitems/",
+                                          issue_id, "/comments", self.api_version,
+                                          "-preview.3")
+        return self._request("GET", url, headers=headers, auth=self.auth)
+
+    def add_comment(self, issue_id, body):
+        headers = {"Content-type": "application/json"}
+        url = "{0}{1}{2}{3}{4}{5}".format(self.base_url, "wit/workitems/",
+                                          issue_id, "/comments", self.api_version,
+                                          "-preview.3")
+        return self._request("POST", url, headers=headers, auth=self.auth, json=body)
+
+    @staticmethod
+    def _request(method, url, **kwargs):
+        return requests.request(method, url, **kwargs).json()
+
+
+class AzureThread(IntegrationThread):
+    """
+        Execute AzureBoards code in a thread!
+
+        Executed from the IssueTracker interface methods.
+    """
+    def post_comment(self):
+        # NOTE: Posting comment is in preview state in API v6.0.
+        comment_body = {
+                "text": markdown2html(self.text())
+            }
+        self.rpc.add_comment(self.bug_id, comment_body)
+
+
+class AzureBoards(IssueTrackerType):
+    """
+        Support for AzureBoards. Requires:
+
+        :base_url: - URL to a AzureBoards project for which we're going to report issues
+        e.g. https://dev.azure.com/{organization}/{project}
+        :api_password: - AzureBoards API token - needs required permissions
```

Comment:
"needs required perissions" - doesn't sound right. It needs "Read & write" permission from the Azure platform.